### PR TITLE
Prettify `test-run.py --help` output

### PR DESCRIPTION
**Pull request by Totktonada**

### CODE REVIEW

**Comment by Totktonada** [comment]
NB: Don't merge before PR #326.

**Comment by Totktonada** [comment]
Rebased after PR #326.

File: lib/options.py [hunk 1]
```diff
@@ -201,135 +286,190 @@ def __init__(self):
...
                 type=int,
-                help="""The number of test run retries after a failure.
-                Default: ${TEST_RUN_RETRIES} or 0. It is also the default value
-                for 'fragile' tests unless the `retries` option is set in the
-                suite.ini config file.""")
+                help=format_help(
+                    """
+                    The number of test run retries after a failure.
+
+                    Default: ${TEST_RUN_RETRIES} or 0. It is also the default
```
Comment (ylobankov) [inline lib/options.py:308]: Let's move `Default: ${TEST_RUN_RETRIES} or 0.` to the end like it's done everywhere.

File: lib/options.py [hunk 2]
```diff
@@ -201,135 +286,190 @@ def __init__(self):
...
                 "--server-start-timeout",
                 dest="server_start_timeout",
                 default=env_int('SERVER_START_TIMEOUT', 90),
                 type=int,
-                help="""Stop the server process if the server starts longer than
-                this amount of seconds. Default: 90 [seconds].""")
+                help=format_help(
+                    """
+                    Stop the server process if the server starts longer than
+                    this amount of seconds. Default: 90 [seconds].
```
Comment (ylobankov) [inline lib/options.py:335]: `Default: 90 [seconds].` should be on the new line.

File: lib/options.py [hunk 3]
```diff
@@ -201,135 +286,190 @@ def __init__(self):
...
+                    Useful for, say, running built-in luatest executable.
 
-                Usage: just source it into a current environment:
+                    Usage: just source it into a current environment:
 
-                ```
-                . <(./test/test-run.py --env)
-                ```
-                """)
+                    ```
```
Comment (ylobankov) [inline lib/options.py:469]: Why not just `$ . <(./test/test-run.py --env)`?

File: lib/options.py (hunk 1 above)
Comment (Totktonada) [reply lib/options.py]: Done. Resolving the discussion as obviously fixed.

File: lib/options.py (hunk 2 above)
Comment (Totktonada) [reply lib/options.py]: Done. Resolving the discussion as obviously fixed.

File: lib/options.py (hunk 3 above)
Comment (Totktonada) [reply lib/options.py]: It feels for me that I should somehow mark a code block. It is not rendered anyhow specific in the `/path/to/test-run.py --help` output, but Markdown is so widespread that I found it acceptable to use it in the plain text for this purpose.

It has not much meaning here, more a matter of taste. I dropped the lines with backticks.

File: lib/options.py [hunk 4]
```diff
@@ -65,134 +75,209 @@ def __init__(self):
...
         parser.add_argument(
                 "--builddir",
                 dest="builddir",
                 default="..",
-                help="""Path to project build directory. Default: ".." """)
+                help=format_help(
+                    """
+                    Path to project build directory.
+
+                    Default: ".."
```
Comment (ylobankov) [inline lib/options.py:239]: 1. The dot is missing at the end of the sentence.
2. IMHO, `Default: ../ .` looks better.

But all this is up to you.

**Review by ylobankov — APPROVED** [review]

File: lib/options.py (hunk 4 above)
Comment (Totktonada) [reply lib/options.py]: I would prefer the following:

```diff
diff --git a/lib/options.py b/lib/options.py
index 834ba91..72dd77b 100644
--- a/lib/options.py
+++ b/lib/options.py
@@ -236,7 +236,11 @@ class Options(object):
                     """
                     Path to project build directory.
 
-                    Default: ".."
+                    Beware: A relative path is resolved against the directory,
+                    where all tests reside. **NOT** a current directory of a
+                    parent shell.
+
+                    Default: .. (parent directory).
                     """))
 
         parser.add_argument(
@@ -257,6 +261,10 @@ class Options(object):
                     """
                     Path to data directory.
 
+                    Beware: A relative path is resolved against the directory,
+                    where all tests reside. **NOT** a current directory of a
+                    parent shell.
+
                     Default: var.
                     """))
```

Applied this change and force-pushed.

Comment (ylobankov) [reply lib/options.py]: I am ok with it.

**Pull request merged**